pipe.go

```diff
@@ -204,6 +209,10 @@ func _newPipe(connFn func() (net.Conn, error), option *ClientOption, r2ps bool)
 				err = r.Error()
 			}
 			if err != nil {
+				if readOnlyCmdIndex >= 0 && i == readOnlyCmdIndex {
```

Comment:
Maybe this way is better and we don't have to introduce the `readOnlyCmdIndex`:
```suggestion
				if init[i][0] == "READONLY" {
```